lib/helpers/utils.js

```diff
@@ -13526,6 +13538,26 @@ export function getPipFrozenTree(
       });
       thoughtLog("Performing poetry install");
       let poetryInstallArgs = ["-m", "poetry", "install", "-n", "--no-root"];
+
+      const pyprojectpath = safeExistsSync(join(basePath, "pyproject.toml"))
+        ? join(basePath, "pyproject.toml")
+        : null;
+      let isPoetryV2 = false;
+      if (pyprojectpath) {
+        try {
+          isPoetryV2 =
+            parsePyProjectTomlFile(pyprojectpath).isPoetryV2 || false;
+        } catch (err) {
```

Comment:
Why do we need a try-catch block here? Are there any exception thrown from `parsePyProjectTomlFile`?